Source/Core/VideoBackends/Vulkan/VKPerfQuery.cpp

```diff
@@ -194,13 +205,15 @@ void PerfQuery::ReadbackQueries(u32 query_count)
     entry.has_value = false;
 
     // NOTE: Reported pixel metrics should be referenced to native resolution
-    m_results[entry.query_type] +=
-        static_cast<u32>(static_cast<u64>(m_query_result_buffer[i]) * EFB_WIDTH /
-                         g_renderer->GetTargetWidth() * EFB_HEIGHT / g_renderer->GetTargetHeight());
+    const u64 native_res_result = static_cast<u64>(m_query_result_buffer[i]) * EFB_WIDTH /
+                                  g_renderer->GetTargetWidth() * EFB_HEIGHT /
+                                  g_renderer->GetTargetHeight();
+    m_results[entry.query_type].fetch_add(static_cast<u32>(native_res_result),
+                                          std::memory_order_relaxed);
   }
 
   m_query_readback_pos = (m_query_readback_pos + query_count) % PERF_QUERY_BUFFER_SIZE;
-  m_query_count -= query_count;
+  m_query_count.fetch_sub(query_count);
```

Comment:
Did you mean to leave off `std::memory_order_relaxed` here?

Reply:
Nope. Thanks for catching that.